src/main/java/io/airlift/units/Duration.java

```diff
@@ -64,9 +64,9 @@ public static Duration succinctDuration(double value, TimeUnit unit)
 
     public Duration(double value, TimeUnit unit)
     {
-        checkArgument(!Double.isInfinite(value), "value is infinite");
+        checkArgument(!Double.isInfinite(value), "value is infinite: %s", value);
```

Comment:
This causes boxing of the value and can have non-negligible performance impact

Reply:
Guava has overloads to avoid boxing, though not for `double`. Since we only have a few uses, better to simply inline the check:
```java
if (Double.isInfinite(value)) {
    throw new IllegalArgumentException("value is infinite: " + value);
}